### DISCUSSION TITLE
AWS EKS SSL Cert pass through

### DISCUSSION THREAD

**Discussion by priteshwatch** [apache/superset#18278] [Q&A / Help]
I was able to setup using the helm service. However, when I tried to use the ssl cert via AWS Cert manager, things dont seemt to work. I am seeing a redirect from https -> http. I would like to only have https exposed. But it seems like superset is redirectly from https to http, not realizing that the https is managed outside. 

service.beta.kubernetes.io/aws-load-balancer-type: nlb
service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:us-east-1:XXXX:certificate/YYYYYY
service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
service.beta.kubernetes.io/aws-load-balancer-ssl-negotiation-policy: "ELBSecurityPolicy-TLS-1-2-2017-01"
external-dns.alpha.kubernetes.io/hostname: superset.domain.com

Is there a way to keep superset on strictly https?

**Comment by priteshwatch** [comment]
I have the following settings in the config

ENABLE_PROXY_FIX = True
PREFERRED_URL_SCHEME = 'https'
FORWARDED_ALLOW_IPS = '*'

There is a 302 redirect from https to http

**Reply by whitney** [reply]
@priteshwatch it seems from your above question you had at least some aspects of your desired SSL setup working. I've just posted [this question](https://github.com/apache/superset/discussions/19785) about getting baseline SSL working. I was wondering if you could share more about the LoadBalancer service definition that you got working?

**Comment by churromorales** [comment]
this is a problem we are encountering as well.

**Comment by nogreintel** [comment]
I also set up the following:
ENABLE_PROXY_FIX = True
and am still  also experiencing the redirect from https to http.

**Comment by nogreintel** [comment]
I explored trying to setup [secure headers](https://docs.gunicorn.org/en/stable/settings.html#secure-scheme-headersl) within gunicorn specifically `secure_scheme_headers='X-FORWARDED-PROTO': 'https'` in a gunicorn.config.py file that would be referenced in the following [script](https://github.com/apache/superset/blob/aee5c9a3ea1006b194319d47c3e7f245e438ce6f/docker/run-server.sh) by adding the line `-c gunicorn.config.py \` to the series of arguments passed into gunicorn. 

This ran without errors, but I couldn't verify if this works or not as there doesn't seem to be a way to configure this via the helm chart, so I tested on a seperate gunicorn server within the same pod.
